lib/caldav_client/tesla.ex

```diff
@@ -14,13 +14,20 @@ defmodule CalDAVClient.Tesla do
 
     auth_middleware =
       case caldav_client.auth do
-        :basic -> Tesla.Middleware.BasicAuth
-        :digest -> Tesla.Middleware.DigestAuth
+        :basic ->
+          {Tesla.Middleware.BasicAuth, credentials}
+
+        :digest ->
+          {Tesla.Middleware.DigestAuth, credentials}
+
+        :bearer ->
+          token = caldav_client.token || raise("Bearer token is missing")
```

Comment:
In general, we would like to avoid using `raise` in favor of `{:error, _}` tuples. However, I see that currently `make_tesla_client` returns `Tesla.Client` directly.

However, I think it would be better to use compile-time checks, just like `@enforce_keys`. Can you please refactor it (probably by introducing new structs) to avoid runtime check here?